Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift

```diff
@@ -33,98 +33,74 @@ public struct HTTPClientResponse: Sendable {
     /// The body of this HTTP response.
     public var body: Body
 
-    /// A representation of the response body for an HTTP response.
-    ///
-    /// The body is streamed as an `AsyncSequence` of `ByteBuffer`, where each `ByteBuffer` contains
-    /// an arbitrarily large chunk of data. The boundaries between `ByteBuffer` objects in the sequence
-    /// are entirely synthetic and have no semantic meaning.
-    public struct Body: Sendable {
-        private let bag: Transaction
-        private let reference: ResponseRef
-
-        fileprivate init(_ transaction: Transaction) {
-            self.bag = transaction
-            self.reference = ResponseRef(transaction: transaction)
-        }
-    }
-
     init(
         bag: Transaction,
         version: HTTPVersion,
         status: HTTPResponseStatus,
         headers: HTTPHeaders
     ) {
-        self.body = Body(bag)
         self.version = version
         self.status = status
         self.headers = headers
+        self.body = Body(TransactionBody(bag))
     }
-}
-
-@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-extension HTTPClientResponse.Body: AsyncSequence {
-    public typealias Element = AsyncIterator.Element
-
-    public struct AsyncIterator: AsyncIteratorProtocol {
-        private let stream: IteratorStream
-
-        fileprivate init(stream: IteratorStream) {
-            self.stream = stream
-        }
-
-        public mutating func next() async throws -> ByteBuffer? {
-            try await self.stream.next()
-        }
-    }
-
-    public func makeAsyncIterator() -> AsyncIterator {
-        AsyncIterator(stream: IteratorStream(bag: self.bag))
+    
+    @inlinable public init(){
+        self.version = .http1_1
+        self.status = .ok
+        self.headers = [:]
+        self.body = Body()
     }
 }
 
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-extension HTTPClientResponse.Body {
-    /// The purpose of this object is to inform the transaction about the response body being deinitialized.
-    /// If the users has not called `makeAsyncIterator` on the body, before it is deinited, the http
-    /// request needs to be cancelled.
-    fileprivate final class ResponseRef: Sendable {
-        private let transaction: Transaction
-
-        init(transaction: Transaction) {
-            self.transaction = transaction
+extension HTTPClientResponse {
+    /// A representation of the response body for an HTTP response.
+    ///
+    /// The body is streamed as an `AsyncSequence` of `ByteBuffer`, where each `ByteBuffer` contains
+    /// an arbitrarily large chunk of data. The boundaries between `ByteBuffer` objects in the sequence
+    /// are entirely synthetic and have no semantic meaning.
+    public struct Body: AsyncSequence, Sendable {
+        public typealias Element = ByteBuffer
+        @usableFromInline typealias Storage = Either<TransactionBody, AnyAsyncSequence<ByteBuffer>>
```

Comment:
TBH, I think the use of `Either` is excessively clever, and it leads to some unclear code (`case a` and `case b`). I'd rather see a specific-case enum for this use-case.